### PR TITLE
Fixes (sea)monkeys, adds lizard hair button, and refactors much of mutantcode

### LINKED ISSUE #2881: Sea monkeys have normal monkeys limbs and no tail
[BUG]

**Describe the bug**
image

**Expected behavior**
Sea monkey limbs and tail

### PR DESCRIPTION
**Pull request by Superlagg**
## About the PR 

Also flashy people arent invisible anymore

Seamonkeys get the right appendages, and humanizing monkeyclones should work properly now. fixes #2881 

Added a button to admin polymorph to set someone's hair override, allowing hairless mutants to have hair. Enables barbering on these hairy lizard monsters.

When a head is severed, its flagged as a transplant, which makes a few vars unable to be changed. Vars like icon, state, and skin color (unless its a lizard's head attached to a lizard, then they can change that color all they want).

Added framework to have mutants have different organs, "hair", torsos, and heads if the human they mutated from is female.

Fixed admin TF applying appearanceholders at the wrong spot when changing mutant race.

Refactored? the way mutants store and define their extra features. Such as:
- Instead of three details being hard-set to their corresponding customization colors and mutantrace dmi, they can now define their own icon and state, and pick which customization color through a var. Couldnt really figure out a decent way to do that *and* let people define a specific color, but its not impossible.
- Removed color flags, they were originally just there cus I was worried about hitting the bit limit thing. Also removed some unused and pointless flags and replaced them with various vars.
- Split normal hair and mutant head-things into two groups: Hair and special hair. This allows lizards and such to have their head things without it messing with their normal hair vals. This also means hairless mutants can be forced to have hair and it is *weird*.
- Moved most skintone flag checks to the mutant initialization procs. Lot easier to keep track of it here.

### CODE REVIEW

**Review by Tarmunora — CHANGES_REQUESTED** [review]
Also, some things to test:
 cloning mutantrace, cloning mutantrace with genetic analysis mode on, cloning with a mutantrace gene in the gene power, cloning a mutantrace with genetic analysis and also having a different mutantrace gene in gene power module.

Cycling a mob through many (all?) mutantraces sequentially

File: code/mob/living/carbon/human/procs/update_icon.dm [hunk 1]
```diff
@@ -828,31 +844,35 @@ var/list/update_body_limbs = list("r_arm" = "stump_arm_right", "l_arm" = "stump_
...
+								human_detail_image.color = AHOLD.customization_second_color
+							if(CUST_3)
+								human_detail_image.color = AHOLD.customization_third_color
+							if(NO_CUST)
 								human_detail_image.color = "#FFFFFF"
-							src.detail_standing_oversuit.overlays += human_detail_image
-							UpdateOverlays(src.detail_standing_oversuit, "detail_oversuit")
-						else // ^^ up here because peoples' bodies turn invisible if it down there with the rest of em
-							UpdateOverlays(null, "detail_oversuit")
+						src.detail_standing_oversuit.overlays += human_detail_image
```
[the remark this answers is not in the document]
Comment (Superlagg) [reply code/mob/living/carbon/human/procs/update_icon.dm:872]: As far as I know, that's how the rest of the proc handles it. I'm also not super sure of the usage, Pali tried explaining it to me at least a few times and I still don't fully get how to use it.

File: code/datums/mutantraces.dm [hunk 2]
```diff
@@ -25,17 +25,17 @@
 	*
 	* For a purely static-icon mutantrace (drawn from a single, non-chunked image), use:
 	*
-	* (IS_MUTANT | HAS_NO_SKINTONE | HAS_NO_HAIR | HAS_NO_EYES | HAS_NO_HEAD | USES_STATIC_ICON)
+	* (NOT_DIMORPHIC | HAS_NO_SKINTONE | HAS_NO_HAIR | HAS_NO_EYES | HAS_NO_HEAD | USES_STATIC_ICON)
 	*
-	* IS_MUTANT prevents the renderer from trying to render a non-existent female sprite, since none of the mutants are dimorphic.
+	* NOT_DIMORPHIC prevents the renderer from trying to render a non-existent female sprite, since none of the mutants are dimorphic.
```
[the remark this answers is not in the document]
Comment (Superlagg) [reply code/datums/mutantraces.dm]: Changed it to actually make sense now.

File: code/mob/living/carbon/human/procs/update_icon.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tarmunora) [reply code/mob/living/carbon/human/procs/update_icon.dm:870]: mood

**Comment by Superlagg** [comment]
Okay:
- Fixed invisible item limbs both being invisible and infecting the rest of the limbs with invisibility
- Made shamblers drop a skintoned human head when beheaded. Still going to look awful until they get chunked.
- Hulk/mutant interaction improved, should maintain the right skintone when combined with mutantraces. leaves you with hulk hair if you lose hulk as a mutant then return to human. 
- Heads should inherit their owner's name properly now. 
- Lizards have the right colors applied to their limb stumps.

**Pull request merged**